phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/apps/ApplicationFilter.java

```diff
@@ -154,6 +222,15 @@ public boolean filter(ApplicationDTO application) {
                     .anyMatch(script -> CollectionUtils.isEmpty(script.getTestingOperatingSystems()));
         }
 
+        /*
+         * If "Show all Operating Systems" is not selected, show only applications that fit to the used operating system
+         */
+        if (!containAllOSCompatibleApplications.getValue()) {
+            result &= application.getScripts().stream()
+                    .anyMatch(script -> script.getCompatibleOperatingSystems()
+                            .contains(operatingSystemFetcher.fetchCurrentOperationSystem()));
```

Comment:
I guess this will crash as well if `getCompatibleOperatingSystems()` returns `null`.